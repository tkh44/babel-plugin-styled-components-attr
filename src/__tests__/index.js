/* eslint-disable jsx-quotes,no-useless-escape,no-template-curly-in-string */
/* eslint-env jest */
const plugin = require('../index')
const babel = require('babel-core')

describe('emotion/babel', () => {
  describe('babel styled component', () => {
    test('something else', () => {
      const basic = `
         const a = \`color: attr(height);\`
      `
      const {code} = babel.transform(basic, {
        plugins: [plugin]
      })
      expect(code).toMatchSnapshot()
    })

    test('empty', () => {
      const basic = `
         styled('input')\`\`
      `
      const {code} = babel.transform(basic, {
        plugins: [plugin]
      })
      expect(code).toMatchSnapshot()
    })

    test('css tag', () => {
      const basic = `css\`
        margin: attr(margin px, 16);
        padding: attr(padding em, 16);
        font-size: attr(fontSize ch, 8);
        width: attr(width %, 95);
        height: attr(height vw, 90);
        display: attr(display, flex);
      \``
      const {code} = babel.transform(basic, {
        plugins: [plugin]
      })
      expect(code).toMatchSnapshot()
    })

    test('basic', () => {
      const basic = `
         styled('input')\`
         height: $\{props => props.height};
         margin: attr(margin);
         padding: attr(padding em, 16);
         color: blue;
        \`
      `
      const {code} = babel.transform(basic, {
        plugins: [plugin]
      })
      expect(code).toMatchSnapshot()
    })



    test('member expression', () => {
      const basic = `styled.input\`
       margin: attr(margin);
       color: #ffffff;
       height: \$\{props => props.height * props.scale\};
       width: attr(width);
       color: blue;
       display: \$\{flex\};
      \``
      const { code } = babel.transform(basic, {
        plugins: [plugin]
      })
      expect(code).toMatchSnapshot()
    })

    test('with value type', () => {
      const basic = `styled('input')\`
        margin: attr(margin px);
      \``
      const { code } = babel.transform(basic, {
        plugins: [plugin]
      })
      expect(code).toMatchSnapshot()
    })

    test('with default value', () => {
      const basic = `styled('input')\`
        margin: attr(margin, 16);
      \``
      const { code } = babel.transform(basic, {
        plugins: [plugin]
      })
      expect(code).toMatchSnapshot()
    })

    test('with value type and default value', () => {
      const basic = `styled('input')\`
        margin: attr(margin px, 16);
      \``
      const { code } = babel.transform(basic, {
        plugins: [plugin]
      })
      expect(code).toMatchSnapshot()
    })

    test('with value type', () => {
      const basic = `styled.input.attrs({ type: 'text' })\`
        margin: attr(margin px);
      \``
      const {code} = babel.transform(basic, {
        plugins: [plugin]
      })
      expect(code).toMatchSnapshot()
    })

    test('kitchen sink', () => {
      const basic = `styled('input')\`
        margin: attr(margin px, 16);
        padding: attr(padding em, 16);
        font-size: attr(fontSize ch, 8);
        width: attr(width %, 95);
        height: attr(height vw, 90);
        display: attr(display, flex);
      \``
      const { code } = babel.transform(basic, {
        plugins: [plugin]
      })
      expect(code).toMatchSnapshot()
    })

    test('all value types', () => {
      const basic = `styled('input')\`
        margin: attr(margin em);
        margin: attr(margin ex);
        margin: attr(margin px);
        margin: attr(margin rem);
        margin: attr(margin vw);
        margin: attr(margin vh);
        margin: attr(margin vmin);
        margin: attr(margin vmax);
        margin: attr(margin mm);
        margin: attr(margin cm);
        margin: attr(margin in);
        margin: attr(margin pt);
        margin: attr(margin pc);
        margin: attr(margin %);
      \``
      const {code} = babel.transform(basic, {
        plugins: [plugin]
      })
      expect(code).toMatchSnapshot()
    })
  })
})
