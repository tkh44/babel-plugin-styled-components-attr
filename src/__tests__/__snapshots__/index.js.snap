// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`emotion/babel babel styled component all value types 1`] = `
"styled('input')\`
        margin: \${function getMargin(props) {
        return props.margin + 'em';
}};
        margin: \${function getMargin(props) {
        return props.margin + 'ex';
}};
        margin: \${function getMargin(props) {
        return props.margin + 'px';
}};
        margin: \${function getMargin(props) {
        return props.margin + 'rem';
}};
        margin: \${function getMargin(props) {
        return props.margin + 'vw';
}};
        margin: \${function getMargin(props) {
        return props.margin + 'vh';
}};
        margin: \${function getMargin(props) {
        return props.margin + 'vmin';
}};
        margin: \${function getMargin(props) {
        return props.margin + 'vmax';
}};
        margin: \${function getMargin(props) {
        return props.margin + 'mm';
}};
        margin: \${function getMargin(props) {
        return props.margin + 'cm';
}};
        margin: \${function getMargin(props) {
        return props.margin + 'in';
}};
        margin: \${function getMargin(props) {
        return props.margin + 'pt';
}};
        margin: \${function getMargin(props) {
        return props.margin + 'pc';
}};
        margin: \${function getMargin(props) {
        return props.margin + '%';
}};
      \`;"
`;

exports[`emotion/babel babel styled component basic 1`] = `
"
styled('input')\`
         height: \${props => props.height};
         margin: \${function getMargin(props) {
         return props.margin;
}};
         padding: \${function getPadding(props) {
         return undefined === props.padding || null === props.padding ? ('16' + 'em') : props.padding;
}};
         color: blue;
        \`;"
`;

exports[`emotion/babel babel styled component css tag 1`] = `
"css\`
        margin: \${function getMargin(props) {
        return undefined === props.margin || null === props.margin ? (\\"16\\" + \\"px\\") : props.margin;
}};
        padding: \${function getPadding(props) {
        return undefined === props.padding || null === props.padding ? (\\"16\\" + \\"em\\") : props.padding;
}};
        font-size: attr(fontSize ch, 8);
        width: \${function getWidth(props) {
        return undefined === props.width || null === props.width ? (\\"95\\" + \\"%\\") : props.width;
}};
        height: \${function getHeight(props) {
        return undefined === props.height || null === props.height ? (\\"90\\" + \\"vw\\") : props.height;
}};
        display: \${function getDisplay(props) {
        return undefined === props.display || null === props.display ? (\\"flex\\" + \\"\\") : props.display;
}};
      \`;"
`;

exports[`emotion/babel babel styled component empty 1`] = `
"
styled('input')\`\`;"
`;

exports[`emotion/babel babel styled component kitchen sink 1`] = `
"styled('input')\`
        margin: \${function getMargin(props) {
        return undefined === props.margin || null === props.margin ? ('16' + 'px') : props.margin;
}};
        padding: \${function getPadding(props) {
        return undefined === props.padding || null === props.padding ? ('16' + 'em') : props.padding;
}};
        font-size: attr(fontSize ch, 8);
        width: \${function getWidth(props) {
        return undefined === props.width || null === props.width ? ('95' + '%') : props.width;
}};
        height: \${function getHeight(props) {
        return undefined === props.height || null === props.height ? ('90' + 'vw') : props.height;
}};
        display: \${function getDisplay(props) {
        return undefined === props.display || null === props.display ? ('flex' + '') : props.display;
}};
      \`;"
`;

exports[`emotion/babel babel styled component member expression 1`] = `
"styled.input\`
       margin: \${function getMargin(props) {
       return props.margin;
}};
       color: #ffffff;
       height: \${props => props.height * props.scale};
       width: \${function getWidth(props) {
       return props.width;
}};
       color: blue;
       display: \${flex};
      \`;"
`;

exports[`emotion/babel babel styled component something else 1`] = `
"
const a = \`color: attr(height);\`;"
`;

exports[`emotion/babel babel styled component with default value 1`] = `
"styled('input')\`
        margin: \${function getMargin(props) {
        return undefined === props.margin || null === props.margin ? ('16' + '') : props.margin;
}};
      \`;"
`;

exports[`emotion/babel babel styled component with value type 1`] = `
"styled('input')\`
        margin: \${function getMargin(props) {
        return props.margin + 'px';
}};
      \`;"
`;

exports[`emotion/babel babel styled component with value type 2`] = `
"styled.input.attrs({ type: 'text' })\`
        margin: \${function getMargin(props) {
        return props.margin + 'px';
}};
      \`;"
`;

exports[`emotion/babel babel styled component with value type and default value 1`] = `
"styled('input')\`
        margin: \${function getMargin(props) {
        return undefined === props.margin || null === props.margin ? ('16' + 'px') : props.margin;
}};
      \`;"
`;
